platform/darwin/src/MGLFeature_Private.h

```diff
@@ -42,4 +42,10 @@ NS_DICTIONARY_OF(NSString *, id) *NSDictionaryFeatureForGeometry(NSDictionary *g
 
 @end
 
+@protocol MGLFeatureCollectionPrivate <NSObject>
```

Comment:
Sounds good. 

`MGLShapeCollectionFeature` declares conformance to `MGLFeaturePrivate` (and raises an exception) if you call the only method in that protocol. I just realized / remembered that this allows for the transformation in `GeometryEvaluator`.